java/vector/src/main/codegen/templates/FixedValueVectors.java

```diff
@@ -162,7 +166,7 @@ public void allocateNew(final int valueCount) {
 
   @Override
   public void reset() {
-    allocationSizeInBytes = INITIAL_VALUE_ALLOCATION;
+    allocationSizeInBytes = INITIAL_VALUE_ALLOCATION * ${type.width};
```

Comment:
`value` is number of elements, not bytes. 

I don't think we want to change the default behavior of initial allocation size...